service/resource/persistentvolume/update.go

```diff
@@ -34,13 +35,66 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState inter
 		return nil
 	}
 
-	_, err = toPV(obj)
+	pv, err := toPV(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	// update state logic
+	switch combinedState := string(rpv.State) + rpv.RecycleState; combinedState {
```

Comment:
I think its good that most of the complexity is in a single place!

Are the case statements in the order the PV is processed? More comments with what each stage does would be useful IMO

Reply:
Yes, case statements are in order of processing

Reply:
Nice and the combined state names are clear which makes it more readable.

Reply:
This reads really well. I think it'd be nice to sort that in alphabetical order to make it easier to realize what combinations are supported.

Reply:
I added description for supported states